src/core/pointcloud/qgspointclouddataprovider.h

```diff
@@ -70,6 +71,16 @@ class CORE_EXPORT QgsPointCloudDataProvider: public QgsDataProvider
       Indexed = 1 << 1 //!< The index is ready to be used
     };
 
+    /**
+     * Point cloud statistics generation state
+     */
+    enum PointCloudStatisticsGenerationState
+    {
+      NotCalculated = 0, //!< Provider has no statistics available
+      Calculating = 1 << 0, //!< Provider try to calculate the statistics
+      Calculated = 1 << 1 //!< The statistics are ready to be used
```

Comment:
When data provider has statistic available from metadata (e.g. EPT), what value would be set - Calculated? if yes, how can we distinguish from whether the stats are coming from provider metadata and whether the stats were calculated by ourselves?

Reply:
It would be set to NotCalculated and the data provider will always fetch statistics from the index. We check for metadata existance using hasStatisticsMetadata function